app/src/lib/api.ts

```diff
@@ -292,10 +292,10 @@ export class API {
         private: private_,
       })
 
-      return parsedResponse<IAPIRepository>(response)
+      return await parsedResponse<IAPIRepository>(response)
     } catch (e) {
-      log.warn(`createRepository: failed with endpoint ${this.endpoint}`, e)
-      throw e
+      log.error(`createRepository: failed with endpoint ${this.endpoint}`, e)
+      throw new Error(`Unable to publish repository. Please check you have an internet connection and try again.`)
```

Comment:
:thumbsup: we can totally check if `e instanceof APIError` here and keep that context around.